diff --git a/pkg/shim/util_unix.go b/pkg/shim/util_unix.go
index d8779c5..468cd20 100644
--- a/pkg/shim/util_unix.go
+++ b/pkg/shim/util_unix.go
@@ -76,7 +76,7 @@ func AdjustOOMScore(pid int) error {
 //
 // TODO: Consider deprecating for CreateSocketAddress
 func SocketAddress(ctx context.Context, socketPath, id string, debug bool) (string, error) {
-	return CreateSocketAddress(ctx, filepath.Join(defaults.DefaultStateDir, "s"), socketPath, id, debug)
+	return CreateSocketAddress(ctx, filepath.Join(defaults.Prefix(defaults.DefaultStateDir), "s"), socketPath, id, debug)
 }
 
 // CreateSocketAddress returns a shim socket address under socketRoot.
@@ -306,7 +306,7 @@ func cleanupSockets(ctx context.Context) {
 	}
 	socketDir, _ := os.Readlink(socketDirLink)
 	if socketDir == "" {
-		socketDir = filepath.Join(defaults.DefaultStateDir, "s")
+		socketDir = filepath.Join(defaults.Prefix(defaults.DefaultStateDir), "s")
 	}
 	if len(socketFlag) > 0 {
 		_ = RemoveSocket("unix://" + socketFlag)
