diff --git a/pkg/shim/util.go b/pkg/shim/util.go
index 613c37f..add057a 100644
--- a/pkg/shim/util.go
+++ b/pkg/shim/util.go
@@ -32,6 +32,7 @@ import (
 	"github.com/containerd/ttrpc"
 	"github.com/containerd/typeurl/v2"
 
+	"github.com/containerd/containerd/v2/defaults"
 	"github.com/containerd/containerd/v2/pkg/atomicfile"
 	"github.com/containerd/containerd/v2/pkg/namespaces"
 	"github.com/containerd/containerd/v2/pkg/protobuf/proto"
@@ -64,6 +65,9 @@ func Command(ctx context.Context, config *CommandConfig) (*exec.Cmd, error) {
 		"-address", config.Address,
 		"-publish-binary", self,
 	}
+	if defaults.PathPrefix != "" {
+		args = append(args, "-prefix", defaults.PathPrefix)
+	}
 	args = append(args, config.Args...)
 	cmd := exec.CommandContext(ctx, config.Runtime, args...)
 	cmd.Dir = config.Path
