diff --git a/cmd/containerd/command/config.go b/cmd/containerd/command/config.go
index 7bbb988..b6f9395 100644
--- a/cmd/containerd/command/config.go
+++ b/cmd/containerd/command/config.go
@@ -126,15 +126,15 @@ func dumpConfig(cliContext *cli.Context) error {
 func platformAgnosticDefaultConfig() *srvconfig.Config {
 	return &srvconfig.Config{
 		Version:          version.ConfigVersion,
-		Root:             defaults.DefaultRootDir,
-		State:            defaults.DefaultStateDir,
+		Root:             defaults.Prefix(defaults.DefaultRootDir),
+		State:            defaults.Prefix(defaults.DefaultStateDir),
 		DisabledPlugins:  []string{},
 		RequiredPlugins:  []string{},
-		StreamProcessors: streamProcessors(),
-		Imports:          []string{defaults.DefaultConfigIncludePattern},
+		StreamProcessors: streamProcessors(defaults.Prefix(defaults.DefaultConfigDir)),
+		Imports:          []string{defaults.Prefix(defaults.DefaultConfigIncludePattern)},
 		Plugins: map[string]any{
 			"io.containerd.server.v1.grpc": map[string]any{
-				"address":               defaults.DefaultAddress,
+				"address":               defaults.Prefix(defaults.DefaultAddress),
 				"max_recv_message_size": defaults.DefaultMaxRecvMsgSize,
 				"max_send_message_size": defaults.DefaultMaxSendMsgSize,
 			},
@@ -142,17 +142,17 @@ func platformAgnosticDefaultConfig() *srvconfig.Config {
 	}
 }
 
-func streamProcessors() map[string]srvconfig.StreamProcessor {
+func streamProcessors(configDir string) map[string]srvconfig.StreamProcessor {
 	const (
 		ctdDecoder = "ctd-decoder"
 		basename   = "io.containerd.ocicrypt.decoder.v1"
 	)
-	decryptionKeysPath := filepath.Join(defaults.DefaultConfigDir, "ocicrypt", "keys")
+	decryptionKeysPath := filepath.Join(configDir, "ocicrypt", "keys")
 	ctdDecoderArgs := []string{
 		"--decryption-keys-path", decryptionKeysPath,
 	}
 	ctdDecoderEnv := []string{
-		"OCICRYPT_KEYPROVIDER_CONFIG=" + filepath.Join(defaults.DefaultConfigDir, "ocicrypt", "ocicrypt_keyprovider.conf"),
+		"OCICRYPT_KEYPROVIDER_CONFIG=" + filepath.Join(configDir, "ocicrypt", "ocicrypt_keyprovider.conf"),
 	}
 	return map[string]srvconfig.StreamProcessor{
 		basename + ".tar.gzip": {
